### ISSUE TITLE
Login doesn't work anymore

### ISSUE THREAD

**Issue by uhertlein** [Diginize/wp-vereinsflieger#16]
As of 9/11/2020 the login via Vereinsflieger doesn't work anymore.

There was an error message regarding:
`src/VereinsfliegerApi/Responses/LoginFailedResponse.php need_2fa` being null where an int was expected.

I've temporarily added a check for that:
```
if (is_null($this->need_2fa)) {
    return 0;
}
```

but login still fails.

The credentials are unchanged and login on the vereinsflieger.de page works fine.

I'm suspecting some vf API change, but am currently unable to debug the issue further.

**Comment by silviokennecke** [comment]
I get the same error. As my appKey does not work at the moment maybe you could verify something for me.

Please make the following HTTP requests and tell me what the second request returns.

```bash
curl --location --request GET 'https://www.vereinsflieger.de/interface/rest/auth/accesstoken'

curl --location --request POST 'https://www.vereinsflieger.de/interface/rest/auth/signin' \
--header 'Content-Type: application/x-www-form-urlencoded' \
--data-urlencode 'username=YOUR-VF-EMAIL' \
--data-urlencode 'password=YOUR-MD5-HASHED-PASSWORD' \
--data-urlencode 'cid=YOUR-CID' \
--data-urlencode 'appkey=YOUR-APP-KEY' \
--data-urlencode 'accesstoken=YOUR-ACCESS-TOKEN'
```

**Comment by silviokennecke** [comment]
I assume that Vereinsflieger has a bug which causes that the app keys are not accepted. I filed a support case. Probably they can help. Their hotline is currently busy.

**Comment by silviokennecke** [comment]
Anyway: **Todo**
Handle invalid app key responses.

**Comment by silviokennecke** [comment]
I just spoke with the Vereinsflieger support. They recently added some more restrictions to API. They will look into it and get back to me. Probably one of the limits is exceeded and this is why the appKey was blocked temporarily.

**Comment by uhertlein** [comment]
I just tried the two 'curl' commands:
- first one successfully returns a valid `accesstoken`
- second one returns an error `{"httpstatuscode":403,"error":"Appkey invalid!"}`

Great to hear that you're in contact with VF support.
After unsuccessfully trying to login I also got a 'Too many attempts' HTTP error (304?) so that might be an issue.

However, I'm unaware of an unusual amount of login attempts from our site, so it'd be good to know what those limits are.

Thanks for looking into that.

**Comment by silviokennecke** [comment]
I got a the log my appKey. This states that some servers try to login via Brute Force. Please try adding a captcha to you login form to prevent this from happening again.

I will update the faq to name this issue.

**Comment by silviokennecke** [comment]
@uhertlein To re-enable your app key please contact the Vereinsflieger support. Please first take actions to prevent attacks on your login form - e.g. by using a captcha. I can recommend this plugin: https://wordpress.org/plugins/google-captcha/

**Comment by silviokennecke** [comment]
Fixed in version v1.0.10
